### ISSUE TITLE
MSSQL is not supported when creating application rule via azurerm_firewall_policy_rule_collection_group

### ISSUE THREAD

**Issue by juicybaba** [hashicorp/terraform-provider-azurerm#17666]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.9

### AzureRM Provider Version

2.99

### Affected Resource(s)/Data Source(s)

azurerm_firewall_policy_rule_collection_group

### Terraform Configuration Files

```hcl
resource "azurerm_firewall_policy_rule_collection_group" "example" {
  name               = "example-fwpolicy-rcg"
  firewall_policy_id = azurerm_firewall_policy.example.id
  priority           = 500
  application_rule_collection {
    name     = "app_rule_collection1"
    priority = 500
    action   = "Deny"
    rule {
      name = "app_rule_collection1_rule1"
      protocols {
        type = "Mssql"
        port = 1433
      }
      source_addresses  = ["10.0.0.1"]
      destination_fqdns = [".microsoft.com"]
    }
  }
}
```

### Debug Output/Panic Output

```shell
expected application_rule_collection.0.rule.0.protocols.0.type to be one of [Http Https], got Mssql
```

### Expected Behaviour

application rule can be created with protocol Mssql and port 1433 via `azurerm_firewall_policy_rule_collection_group`

### Actual Behaviour

application rule **can't** be created with protocol Mssql and port 1433 via `azurerm_firewall_policy_rule_collection_group`

### References

As per document, `Mssql` is support by [azurerm_firewall_application_rule_collection](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/firewall_application_rule_collection#type)  but not supported by [azurerm_firewall_policy_rule_collection_group](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/firewall_policy_rule_collection_group#type), but customers should be able create application rule with both above resources.

**Comment by xiaxyi** [comment]
@henryxuteck I do see `mssql` is not supported in [azurerm_firewall_policy_rule_collection_group](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/firewall_policy_rule_collection_group#type). I need to confirm this feature at business level to see if we should add it.

will let you know.

**Comment by xiaxyi** [comment]
@henryxuteck I'll work on the fix, will update accordingly

**Comment by henryxuteck** [comment]
> @henryxuteck I do see `mssql` is not supported in [azurerm_firewall_policy_rule_collection_group](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/firewall_policy_rule_collection_group#type). I need to confirm this feature at business level to see if we should add it.
> 
> 
> 
> will let you know.

@xiaxyi any updates? 😁

**Comment by xiaxyi** [comment]
Thanks @henryxuteck for the follow-ups, I'm confirming internally, may need more times. I'll update once there is any.

**Comment by xiaxyi** [comment]
Hi @henryxuteck I added the `Mssql` in the supported protocols, you can track the status in the mentioned pr. Feel free to let me know if there is anything needed, Thanks.

**Comment by henryxuteck** [comment]
> Hi @henryxuteck I added the `Mssql` in the supported protocols, you can track the status in the mentioned pr. Feel free to let me know if there is anything needed, Thanks.

@xiaxyi awesome, thanks

**Comment by juicybaba** [comment]
@xiaxyi It seems like this issue is fixed via #18746, can you double confirm?

**Comment by xiaxyi** [comment]
@juicybaba yes, it's fixed via #18746 . Can you try testing Mssql from your side and see if you can get it work, then I think it's ok for us to close this issue.

**Comment by juicybaba** [comment]
> @juicybaba yes, it's fixed via #18746 . Can you try testing Mssql from your side and see if you can get it work, then I think it's ok for us to close this issue.

Hi @xiaxyi , the fix looks good, this issue can be closed. Thanks.
